### PR TITLE
Optimize flow analysis assembly

### PR DESCRIPTION
**Pull request by genlu**
@tmat @mavasani 

Do I need to add an assembly entry in OptProf.json as well?

### CODE REVIEW

**Comment by tmat** [comment]
> Do I need to add an assembly entry in OptProf.json as well?

I don't think so - as long as the assembly is contained in Roslyn.VisualStudio.Setup.vsix

**Comment by genlu** [comment]
@tmat hmm, I assumed the flow analysis assembly should show up in  `Optprof data files` of roslyn build artifacts. But it's not in the build I queued:
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=2823803

Any idea how to verify this change?

**Comment by genlu** [comment]
@tmat I made the fix as discussed offline. Could you please take another look?

File: eng/targets/VisualStudio.targets [hunk 1]
```diff
@@ -69,12 +69,34 @@
...
+      <_AssetsNotToOptimize Include="@(_AssetsByPackageId)" />
+      <_AssetsNotToOptimize Remove="@(_PackageToOptimize)" />
+
+      <_AllAssetsToInclude Include="@(_AssetsToOptimize)">
+        <Optimization>true</Optimization>
+      </_AllAssetsToInclude>
+
+      <_AllAssetsToInclude Include="@(_AssetsNotToOptimize)">
+        <Optimization>false</Optimization>
+      </_AllAssetsToInclude>
```
Comment (genlu) [inline eng/targets/VisualStudio.targets]: @tmat I know this looks ridicules, but it's what I could figure out with my poor MSBuild skills. There's gotta be a better way, right...?

**Comment by genlu** [comment]
So I have confirmed in master that this does enable optprof data collection for FlowAnalysis dll. But further verification is block there because the missing optprof data with EE.ResultProvider. Maybe we should just merge this into 16.2?

@jinujoseph @vatsalyaagrawal Need approval.

**Comment by jinujoseph** [comment]
Approved to merge to test if the fix will resolve the optProf . should be reverted if not fixed

**Comment by genlu** [comment]
@RikkiGibson I will merge this to dev16.2-vs-deps for testing purpose. Could you please halt the codeflow until I confirmed the fix works?

**Comment by genlu** [comment]
@RikkiGibson nvm, all RPS bugs are moved to 16.3

**Pull request closed without merging**